scripts/sct_analyze_lesion.py

```diff
@@ -104,410 +101,123 @@ def get_parser():
+    _, arr_ctl, arr_ctl_der, _ = get_centerline(im_seg, param=ParamCenterline(), verbose=1)
+    x_centerline_deriv, y_centerline_deriv, z_centerline_deriv = arr_ctl_der
 
-        # copy seg image
-        if self.fname_sc is not None:
-            sct.copy(self.fname_sc, self.tmp_dir)
-            self.fname_sc = ''.join(extract_fname(self.fname_sc)[1:])
+    angle_correction = np.full_like(np.empty(nz), np.nan, dtype=np.double)
 
-        # copy ref image
-        if self.fname_ref is not None:
-            sct.copy(self.fname_ref, self.tmp_dir)
-            self.fname_ref = ''.join(extract_fname(self.fname_ref)[1:])
+    # loop across x_centerline_deriv
+    # (instead of [min_z_index, max_z_index], which could vary after interpolation)
+    for iz in range(x_centerline_deriv.shape[0]):
+        # normalize the tangent vector to the centerline (i.e. its derivative)
+        vect = np.array([x_centerline_deriv[iz] * px, y_centerline_deriv[iz] * py, pz])
+        norm = np.linalg.norm(vect)
+        tangent_vect = vect / norm
 
-        # copy registered template
-        if self.path_template is not None:
-            sct.copy(self.path_levels, self.tmp_dir)
-            self.path_levels = ''.join(extract_fname(self.path_levels)[1:])
+        # compute the angle between the normal vector of the plane and the vector z
+        angle = np.arccos(np.vdot(tangent_vect, np.array([0, 0, 1])))
+        angle_correction[iz] = math.degrees(angle)
 
-            self.atlas_roi_lst = []
-            for fname_atlas_roi in os.listdir(self.path_atlas):
-                if fname_atlas_roi.endswith('.nii.gz'):
-                    tract_id = int(fname_atlas_roi.split('_')[-1].split('.nii.gz')[0])
-                    if tract_id < 36:  # Not interested in CSF
-                        sct.copy(os.path.join(self.path_atlas, fname_atlas_roi), self.tmp_dir)
-                        self.atlas_roi_lst.append(fname_atlas_roi)
+    return angle_correction
 
-        os.chdir(self.tmp_dir)  # go to tmp directory
+
+def analyze_lesion(fname_mask, fname_voi, fname_ref=None, path_template=None, path_ofolder="./analyze_lesion", verbose=1):
+    """
+    Analyze lesions or tumours by computing statistics on binary mask.
+
+    :param fname_mask: Lesion binary mask filename.
+    :param fname_voi: Volume of interest binary mask filename.
+    :param fname_ref: Image filename from which to extract average values within lesions.
+    :param path_template: Path to folder containing the atlas/template registered to the anatomical image.
+    :param path_ofolder: Output folder.
+    :param verbose: Verbose.
+    :return: XX
+    """
+    im_mask, im_voi = Image(fname_mask), Image(fname_voi)
+
+    # Check if input data is binary and not empty
+    check_binary(im_mask.data, fname=fname_mask, verbose=verbose)
+    check_binary(im_voi.data, fname=fname_voi, verbose=verbose)
+
+    # re-orient image to RPI
+    logger.info("Reorient the image to RPI, if necessary...")
```

Comment:
as per [convention](https://github.com/neuropoly/spinalcordtoolbox/wiki/logging) we use printv in CLI and logger in APIs. However, i'm thinking maybe we should move `analyze_lesion` under `spinalcordtoolbox`, and maybe make it more "general", i.e., `analyze_binary_objects`. I see a lot or resemblance in terms of thematics with `spinalcordtoolbox.process_seg`, so maybe we should move it there?